src/CBT.NuGet/Internal/FileUtilities.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CBT.NuGet.Internal
+{
+    class FileUtilities
+    {
+        // Need to find a proper directory copy routine.
+        public static void DirectoryCopy(string sourceDirName, string destDirName, bool copySubDirs, bool overWriteFiles)
+        {
+            if (!Directory.Exists(sourceDirName))
+            {
+                throw new DirectoryNotFoundException("Source Directory does not exist or could not be found: " + sourceDirName);
+            }
+            DirectoryInfo dir = new DirectoryInfo(sourceDirName);
+            DirectoryInfo[] dirs = dir.GetDirectories();
```

Comment:
Use `EnumerateDirectories` instead and `EnumerateFiles` instead of `GetFiles` below.  We should probably revisit this method and see if Async copies or parallelism would help here.